src/solvers/refinement/string_refinement.cpp

```diff
@@ -1665,6 +1715,9 @@ bool string_refinementt::is_axiom_sat(
   case decision_proceduret::resultt::D_UNSATISFIABLE:
     return false;
   default:
-    throw "failure in checking axiom";
+    error() << "failure in checking axiom" << eom;
```

Comment:
My view is we should initially at least always dump an error report when one is violated, even if caught